redisson/src/main/java/org/redisson/api/RPermitExpirableSemaphore.java

```diff
@@ -102,20 +102,44 @@ public interface RPermitExpirableSemaphore extends RExpirable, RPermitExpirableS
     void release(String permitId);
     
     /**
-     * Returns amount of available permits.
+     * Returns number of available permits.
      *
-     * @return number of permits
+     * @return number of available permits
      */
     int availablePermits();
 
     /**
-     * Tries to set number of permits.
+     * Returns the number of permits.
+     *
+     * @return number of permits
+     */
+    int getPermits();
+
+    /**
+     * Returns the number of claimed permits.
+     *
+     * @return number of claimed permits
+     */
+    int claimedPermits();
+
+    /**
+     * Tries to set the initial number of available permits.
      *
      * @param permits - number of permits
      * @return <code>true</code> if permits has been set successfully, otherwise <code>false</code>.  
      */
     boolean trySetPermits(int permits);
-    
+
+    /**
+     * Sets the number of permits such that the number of claimed
+     * permits plus the number of available permits is equal to the provided value.
+     * If the number of permits has not been set yet, it will be set to the provided value.
+     *
+     * @param permits - number of permits to use as the maximum
```

Comment:
Please correct the description. Since we use delta here. Also what is the point to return this number? Where might we need it? Can't we just return `void`?

Reply:
It may be helpful information to clients to know if there was a zero or non-zero number of permits added. It's not strictly required but it could make some client's testing and edge case handling a bit easier.

Reply:
I think void is better return type for this setter

Reply:
Sounds good 👍